src/format.ts

```diff
@@ -42,6 +42,42 @@ class FormattedString implements Stringable {
     this.entities = entities;
   }
 
+  /**
+   * Replaces matches of a pattern with a replacement string
+   *
+   * @param searchValue The pattern to search for
+   * @param replaceValue The string to replace the pattern with
+   * @returns A new `FormattedString` with replacements
+   */
+  replace(searchValue: string | RegExp, replaceValue: string): FormattedString {
+    const newText = this.text.replace(searchValue, replaceValue);
+
+    const newEntities = this.entities.map(entity => {
+      const newEntity = { ...entity };
+
+      if (typeof searchValue === 'string') {
+        const index = this.text.indexOf(searchValue);
+
+        if (index !== -1 && index < newEntity.offset + newEntity.length) {
+          newEntity.offset = newEntity.offset + replaceValue.length - searchValue.length;
```

Comment:
Thanks for the PQ!

Please consider this counterexample test case in your next commit.

```ts
bot.command('start', async ctx => {
  const msg = fmt`I like the grammY ${code('framework')} very much`;

  const newMsg = msg.replace('framework', 'library');

  await ctx.replyFmt(newMsg);
});
```
Result:
<img width="408" alt="Screenshot 2024-06-02 at 10 45 35 AM" src="https://github.com/grammyjs/parse-mode/assets/9781814/da350529-5ea1-4db2-a88b-9eebcdb2988d">